infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/configurator/OpenShiftWorkspaceServiceAccountConfigurator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2012-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.project.configurator;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Sets;
+import java.util.Collections;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.spi.NamespaceResolutionContext;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.configurator.NamespaceConfigurator;
+import org.eclipse.che.workspace.infrastructure.openshift.OpenShiftClientFactory;
+import org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftWorkspaceServiceAccount;
+
+public class OpenShiftWorkspaceServiceAccountConfigurator implements NamespaceConfigurator {
+
+  private final OpenShiftClientFactory clientFactory;
+
+  private final String serviceAccountName;
+  private final Set<String> clusterRoleNames;
+
+  @Inject
+  public OpenShiftWorkspaceServiceAccountConfigurator(
```

Comment:
is it somehow different from WorkspaceServiceAccountConfigurator?

Reply:
this one is using `OpenShiftWorkspaceServiceAccount`, the `WorkspaceServiceAccountConfigurator` is using `KubernetesWorkspaceServiceAccount`.

While it could be done in more generic way, I'm not sure it's worth it. WDYT?

Reply:
I'm fine with that.